test/1_stdlib/Float.swift

```diff
@@ -222,13 +222,24 @@ func checkQNaN(_ qnan: TestFloat) {
   _precondition(qnan.floatingPointClass == .quietNaN)
 }
 
+func checkSNaN(_ snan: TestFloat) {
+  checkNaN(snan)
+// FIXME: i386 doesn't fully support signaling NaN.
```

Comment:
Yup, it ends up being a ~30% perf win for smallish math functions.  The only question is how much anyone cares about i386.